web/src/containers/EncounterList/hooks.ts

```diff
@@ -24,116 +39,109 @@ export function useEncounterList(
-                          ? [
-                                `ge${formatFHIRDateTime(dateFilterValue[0])}`,
-                                `le${formatFHIRDateTime(dateFilterValue[1])}`,
-                            ]
-                          : undefined,
-                  },
-              )
-            : success(undefined);
-
-        return mapSuccess(response, (bundle) => {
-            const sourceMap = bundle
-                ? extractBundleResources(bundle)
-                : { Encounter: [], Patient: [], Practitioner: [], PractitionerRole: [] };
-
-            const encounters = sourceMap.Encounter;
-            const patients = sourceMap.Patient;
-            const practitioners = sourceMap.Practitioner;
-            const practitionerRoles = sourceMap.PractitionerRole;
-
-            return encounters.map((encounter) => {
-                const patient: WithId<Patient> | undefined = patients.find(
-                    (p) => p.id === encounter.subject?.id,
-                );
-                const practitionerRole = practitionerRoles.find(
-                    (pR) => pR.id === encounter.participant?.[0]!.individual?.id,
-                );
-
-                const practitioner: WithId<Practitioner> | undefined = practitioners.find(
-                    (p) => p.id === practitionerRole?.practitioner?.id,
-                );
-
-                return {
-                    id: encounter.id,
-                    patient,
-                    practitioner,
-                    status: encounter.status,
-                    date: encounter?.period?.start,
-                    humanReadableDate:
-                        encounter?.period?.start && formatHumanDateTime(encounter?.period?.start),
-                };
-            });
+    const patientFilterValue = filterValues?.[0]?.value ?? undefined;
+    const practitionerFilterValue = filterValues?.[1]?.value ?? undefined;
+    const dateFilterValue = filterValues?.[2]?.value ?? undefined;
+
+    const dateParameter = dateFilterValue
+        ? [
+              `ge${formatFHIRDateTime(dateFilterValue[0])}`,
+              `le${formatFHIRDateTime(dateFilterValue[1])}`,
+          ]
+        : undefined;
+
+    const queryParameters = {
+        ...searchParams,
+        _include: [
+            'Encounter:subject',
+            'Encounter:participant:PractitionerRole',
+            'PractitionerRole:practitioner:Practitioner',
+        ],
+        'participant-display': practitionerFilterValue,
+        'subject:Patient.name': patientFilterValue,
+        date: dateParameter,
+        _sort: '-_lastUpdated',
+    };
+
+    const { resourceResponse, pagerManager, handleTableChange, pagination } = usePagerExtended<
+        Encounter | PractitionerRole | Practitioner | Patient,
+        EncounterListFilterValues
+    >('Encounter', queryParameters, debouncedFilterValues);
+
+    const encounterData = mapSuccess(resourceResponse, (bundle) => {
```

Comment:
I would prefer to wrap `encounterData` with `useMemo` here because it does quite many calculations